src/PosthogAnalytics.ts

```diff
@@ -218,29 +235,56 @@ export class PosthogAnalytics {
     if (this.anonymity == Anonymity.Pseudonymous) {
       // Check the user's account_data for an analytics ID to use. Storing the ID in account_data allows
       // different devices to send the same ID.
+      let analyticsID = await this.getAnalyticsId(client);
       try {
-        const accountData = await client.getAccountDataFromServer(
-          PosthogAnalytics.ANALYTICS_EVENT_TYPE
-        );
-        let analyticsID = accountData?.id;
-        if (!analyticsID) {
+        if (!analyticsID && !widget) {
+          // only try setting up a new analytics ID in the standalone app.
+
           // Couldn't retrieve an analytics ID from user settings, so create one and set it on the server.
           // Note there's a race condition here - if two devices do these steps at the same time, last write
           // wins, and the first writer will send tracking with an ID that doesn't match the one on the server
           // until the next time account data is refreshed and this function is called (most likely on next
           // page load). This will happen pretty infrequently, so we can tolerate the possibility.
           analyticsID = analyticsIdGenerator();
-          await client.setAccountData(
-            PosthogAnalytics.ANALYTICS_EVENT_TYPE,
-            Object.assign({ id: analyticsID }, accountData)
-          );
+          await this.setAnalyticsId(analyticsID, client);
         }
-        this.posthog.identify(analyticsID);
       } catch (e) {
         // The above could fail due to network requests, but not essential to starting the application,
         // so swallow it.
         logger.log("Unable to identify user for tracking" + e.toString());
       }
+      if (analyticsID) {
+        this.posthog.identify(analyticsID);
```

Comment:
This looks like it could leak an EW analytics ID to a different posthog instance?

Reply:
The plan is so that element web and element call use the same analytics id. That mirrors the behaviour of using the standalone element call client. (There the analytics id would be loaded from the account data if already available)
The url parameters from where the id is passed are element call specific. So no other widget will have access to the id.

Reply:
Don't they have separate configs and thus could point at different posthog instances?

Reply:
This is now fixed by only sending a hash of `'ec' + accountAnalyticsId + client.getUserId()` to posthog.